### PR TITLE
Issue #459: Start locale refactoring

### PR DESCRIPTION
**Pull request by lasconic**
- [ ] Fixes #
- [x] Tests added/updated.
- [x] `./check.sh` run and eventual issues fixed.
- [x] Updated CONTRIBUTORS.md, if not already done in the past.

### CODE REVIEW

**Comment by lasconic** [comment]
Did you envision something like this ? 
So move all ìf` statement in a dictionary of string->function and call it from `last template handler` ?

Then maybe call it from the main code and not each locale.

It's mainly a lot of copy paste to do... so if the approach looks right I will just go ahead and do it.

**Comment by lasconic** [comment]
Also, I wonder... do we keep one liner in last_template_handler ?

**Comment by BoboTiG** [comment]
Yeah, I had that kind of refactoring in minde 👍 

I think one-liners should go to the `templates_multi` dict, WDYT?

File: scripts/lang/fr/template_handlers.py [hunk 1]
```diff
@@ -0,0 +1,350 @@
...
+
+
+def lookup_template(tpl: str) -> bool:
+    return tpl in template_mapping
+
+
+def render_template(template: Tuple[str, ...], locale: str, word: str = "") -> str:
+    tpl, *parts = template
+    data = extract_keywords_from(parts)
+    return template_mapping[tpl](tpl, parts, data, word, locale)
```
Comment (BoboTiG) [inline scripts/lang/fr/template_handlers.py:350]: We could go witout the `tpl` argument. Same for `locale` as it will not change.

**Review by BoboTiG — COMMENTED** [review]
Ideally, you can move all docstests to their own function.
And function should sorted by name too :)

File: scripts/lang/fr/template_handlers.py (hunk 1 above)
Comment (lasconic) [reply scripts/lang/fr/template_handlers.py:350]: tpl is needed I'm afraid. Locale can go. Thanks

**Comment by lasconic** [comment]
> Yeah, I had that kind of refactoring in minde 👍
> 
> I think one-liners should go to the `templates_multi` dict, WDYT?

Unfortunately some one liner need `data` which is not available in templates_multi

File: scripts/lang/fr/template_handlers.py (hunk 1 above)
Comment (BoboTiG) [reply scripts/lang/fr/template_handlers.py:350]: Ah yes, because several templates are using the same function. That makes sense.

**Comment by lasconic** [comment]
This is tedious as hell... but I believe fr is done. I don't look forward doing english :) I will wait for comments first.

**Comment by BoboTiG** [comment]
That way better now 💪

**Comment by BoboTiG** [comment]
For me, it is good to be merged as-is. Can I process or do you plan other changes?

**Comment by lasconic** [comment]
Go for it. I will do it form the other languages next. I don't really like the defaultdict in all the tests... do you have a better idea ?

**Comment by BoboTiG** [comment]
Yeah me too. I was thinking of passing it as a keyword only so that we could get rid of `defaultdict(str)` when it is not needed. And for cases when it is needed, just pas a `dict` and not a `defaultdict`.

Example (not tested):
```python
def render_acronyme(tpl: str, parts: List[str], data: Dict[str, str] = None) -> str:
    """
    >>> render_acronyme("acronyme", ["fr"])
    '<i>(Acronyme)</i>'
    >>> render_acronyme("acronyme", ["en"], data={"de":"light-emitting diode", "m":"oui"}))
```
But then the linter will cry ...

Another thing: maybe the `tpl` can be keyword only  too. It will prevent setting it for all test cases.

**Comment by lasconic** [comment]
I tried this... Linter not happy...  and we often use dict[key] notation for missing key, passing a dict by keyword will not work... 

Maybe a more robust solution... get rid of parts and put them in data instead as key "1", "2" etc... just like mediawiki is doing I believe...

**Comment by BoboTiG** [comment]
That would be the good move, yes. Let's do that later (and again a big refactoring to handle haha).

**Review by BoboTiG — APPROVED** [review]

**Pull request merged**